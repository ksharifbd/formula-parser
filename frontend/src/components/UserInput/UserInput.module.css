.UserInput__input {
    font-size: 1.4rem;
}