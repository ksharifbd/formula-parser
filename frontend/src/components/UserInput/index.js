import React from 'react';
import PropTypes from 'prop-types';

import MediaText from '../MediaText';
import classes from './UserInput.module.css';

const UserInput = ({label, value, onChange}) => (
    <MediaText
        marginRight='6rem'
        left={<label>{label}</label>}
        right={<input
                type='text' 
                value={value}
                className={classes.UserInput__input}
                onChange={onChange} />} />
);

UserInput.propTypes = {
    label: PropTypes.string.isRequired,
    value: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
}

export default UserInput;
