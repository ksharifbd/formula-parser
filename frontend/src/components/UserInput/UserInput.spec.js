import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import UserInput from './';

describe('User Input', () => {
    const tree = shallow(
        <UserInput
            label='x'
            name='x'
            value='2'
            onChange={jest.fn}
        />
    );

    it('should render correctly', () => {
        expect(toJson(tree)).toMatchSnapshot();
    });
});
