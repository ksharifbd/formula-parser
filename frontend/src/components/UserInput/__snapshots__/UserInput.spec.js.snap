// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Input should render correctly 1`] = `
<MediaText
  left={
    <label>
      x
    </label>
  }
  marginRight="6rem"
  right={
    <input
      className="UserInput__input"
      name="x"
      onChange={[Function]}
      type="text"
      value="2"
    />
  }
/>
`;
